sympy/integrals/tests/test_transforms.py

```diff
@@ -705,6 +710,20 @@ def test_laplace_transform():
         f(w), w, t, plane=0) == InverseLaplaceTransform(f(w), w, t, 0)
     assert LT(f(t)*g(t), t, s) == LaplaceTransform(f(t)*g(t), t, s)
 
+    # The following test confirms that issue #17098 is solved
+    F, _, _ = laplace_transform(diff(y(t), t, 2)
+                                + y(t)
+                                - 4*(Heaviside(t - 1)
+                                - Heaviside(t - 2)), t, s)
+    G = laplace_transform_replace_ic(F, y, Y, t, s, [1, 0])
+    assert (G == s**2*Y(s) - s + Y(s) - 4*exp(-s)/s + 4*exp(-2*s)/s)
+
+    # laplace_transform covers lists and replacements
+    assert laplace_transform([diff(z(t), t, 2)-a*z(t)-b*y(t),
+                              diff(y(t), t)-c*z(t)-d*y(t)], t, s,
+                             {z: (Z, [z0, z1, 0]), y: (Y, [y0, 0, 0])}
```

Comment:
I think it's better to have the user provide `z(t)` and `Z(s)` rather than `z` and `Z`. Note that it's quite a common use pattern to define the python variable `z` as the expression `z(t)` i.e.:
```python
In [8]: t, s = symbols('t, s')

In [9]: z = Function('z')(t)

In [10]: Z = Function('Z')(s)

In [11]: z
Out[11]: z(t)

In [12]: Z
Out[12]: Z(s)
```

Reply:
Done.